### PR TITLE
Upgrading Braze Web SDK to v4

### PR DESCRIPTION
**Pull request by spvismaya**
## Summary
- Upgrading Braze Web SDK to the latest v4.2.1

## Testing Plan
- Updated the test suite as needed and they all passed

## Reference Issue
- Closes https://go.mparticle.com/work/REPLACEME

### CODE REVIEW

File: src/BrazeKit-dev.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ window.appboy = require('@braze/web-sdk');
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-var name = 'Appboy',
+var name = 'Braze',
```
Comment (spvismaya) [inline src/BrazeKit-dev.js:17]: Not sure what this controls, please confirm if this should / should not be changed

**Comment by spvismaya** [comment]
Please let me know if anything else needs to upgrade SDK to v4. More details of v3 to v4 upgrade can be found [here](https://github.com/braze-inc/web-sdk/blob/develop/Public/UPGRADE_GUIDE.md). Thank you!

**Comment by rmi22186** [comment]
Hi @spvismaya - it looks like a few commits that we brought into the braze web kit have caused some conflicts.  I should be able to resolve them pretty quickly, but it may make more sense in that case for me to close this, cherrypick your commit, and push, then i'll add you as a reviewer.  How does that sound?  I'll add some comments here though as an FYI.

File: src/BrazeKit-dev.js (hunk 1 above)
Comment (rmi22186) [reply src/BrazeKit-dev.js:17]: This is something internal for us that we check against what is in our database, and it should not be changed, thanks for the question though!

Comment (spvismaya) [reply src/BrazeKit-dev.js:17]: Thanks for confirming! I will revert the change shortly

**Comment by spvismaya** [comment]
> Hi @spvismaya - it looks like a few commits that we brought into the braze web kit have caused some conflicts. I should be able to resolve them pretty quickly, but it may make more sense in that case for me to close this, cherrypick your commit, and push, then i'll add you as a reviewer. How does that sound? I'll add some comments here though as an FYI.

Sure, I am okay with whichever approach works better for you!

**Comment by rmi22186** [comment]
closing this as this has PR has been replicated and expanded upon in PR https://github.com/mparticle-integrations/mparticle-javascript-integration-braze/pull/12

**Pull request closed without merging**